plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -237,35 +239,56 @@ public static ReaderPageSource createPageSource(
             messageColumn = getColumnIO(fileSchema, requestedSchema);
 
             Map<List<String>, ColumnDescriptor> descriptorsByPath = getDescriptors(fileSchema, requestedSchema);
-            TupleDomain<ColumnDescriptor> parquetTupleDomain = options.isIgnoreStatistics()
-                    ? TupleDomain.all()
-                    : getParquetTupleDomain(descriptorsByPath, effectivePredicate, fileSchema, useColumnNames);
-
-            TupleDomainParquetPredicate parquetPredicate = buildPredicate(requestedSchema, parquetTupleDomain, descriptorsByPath, timeZone);
+            List<ParquetDomainAndPredicate> domainAndPredicates;
+            Optional<FilterPredicate> filterPredicate;
+            if (!options.isIgnoreStatistics()) {
+                domainAndPredicates = disjunctTupleDomains.stream()
+                        .map(domain -> {
+                            TupleDomain<ColumnDescriptor> parquetTupleDomain = getParquetTupleDomain(descriptorsByPath, domain, fileSchema, useColumnNames);
+                            TupleDomainParquetPredicate parquetPredicate = buildPredicate(requestedSchema, parquetTupleDomain, descriptorsByPath, timeZone);
+                            return new ParquetDomainAndPredicate(parquetTupleDomain, parquetPredicate);
+                        })
+                        .collect(toImmutableList());
+
+                filterPredicate = domainAndPredicates.stream()
+                        .map(ParquetDomainAndPredicate::predicate)
+                        .map(parquetPredicate -> parquetPredicate.toParquetFilter(timeZone))
+                        .flatMap(Optional::stream)
+                        // TODO create balanced tree when this is used with high number of filters.
+                        .reduce(FilterApi::or);
```

Comment:
We're not actually using `filterPredicate` in this class, so why not let ParquetReader handle creating this internally ?